proto/api/v1/cloud_source_service.proto

```diff
@@ -0,0 +1,158 @@
+syntax = "proto3";
+
+package v1;
+
+import "api/v1/empty.proto";
+import "api/v1/pagination.proto";
+import "gogoproto/gogo.proto";
+import weak "google/api/annotations.proto";
+
+option go_package = "v1";
+option java_package = "io.stackrox.proto.api.v1";
+
+// CloudSource is an integration which provides a source for discovered
+// clusters.
+message CloudSource {
+  string id = 1;
+  string name = 2;
+  enum Type {
+    UNSPECIFIED = 0;
+    PALADIN_CLOUD = 1;
+    OCM = 2;
+  }
+  Type type = 3;
+
+  message Credentials {
+    string secret = 1 [(gogoproto.moretags) = 'scrub:"always"'];
+  }
+  Credentials credentials = 4;
+
+  bool skip_test_integration = 5;
+
+  oneof Config {
+    PaladinCloudConfig paladin_cloud = 6;
+    OCMConfig ocm = 7;
+  }
+}
+
+// PaladinCloudConfig provides information required to fetch discovered
+// clusters from Paladin Cloud.
+message PaladinCloudConfig {
+  string endpoint = 1 [(gogoproto.moretags) = 'validate:"nolocalendpoint"'];
```

Comment:
nit: on the documentation front, would be nice to add the expected format and that https is assumed (same applies to the OCM stuff).